analysis_driver/pipelines/qc_gatk4.py

```diff
@@ -0,0 +1,643 @@
+        var_filter_command += " -V " + self.raw_indels_vcf
+        var_filter_command += " --filter-expression " + filters
+        var_filter_command += " --filter-name 'INDEL_HARD_FILTER'"
+        variant_filter_status = executor.execute(
+            var_filter_command,
+            job_name='indel_filtration',
+            working_dir=self.exec_dir,
+            mem=16
+        ).join()
+        return variant_filter_status
+
+
+class MergeVariants(GATK4Stage):
+    """Merge multiple vcf files into one."""
+
+    vcf_files = ListParameter()
+    output_vcf_file = Parameter()
+
+    def _run(self):
+        # get the name of the list from the name of the first vcf to avoid overwriting another file
+        base_name = os.path.splitext(os.path.basename(self.vcf_files[0]))[0]
+        vcf_list = os.path.join(self.job_dir, base_name + '.list')
+        with open(vcf_list, 'w') as open_file:
+            for f in self.vcf_files:
+                open_file.write(f + '\n')
+        cmd = self.gatk_picard_cmd('MergeVcfs', self.output_vcf_file, input=vcf_list, reference=None)
+        return executor.execute(
+            cmd,
+            job_name='merge_vqsr',
+            working_dir=self.exec_dir,
+            cpus=1,
+            mem=8
+        ).join()
+
+
+def build_pipeline(dataset):
+    """Build the QC pipeline."""
+
+    def stage(cls, **params):
+        return cls(dataset=dataset, **params)
+
+    # merge fastq to do contamination check
+    merge_fastqs = stage(MergeFastqs)
+    contam = stage(qc.FastqScreen, previous_stages=[merge_fastqs], fq_pattern=merge_fastqs.fq_pattern)
+    blast = stage(qc.Blast, previous_stages=[merge_fastqs], fastq_file=merge_fastqs.fq_pattern.replace('?', '1'))
+
+    # create fastq index then align via scatter-gather strategy
+    fastq_index = stage(FastqIndex)
+    split_bwa = stage(SplitBWA, previous_stages=[fastq_index])
+    merge_bam_dup = stage(MergeBamAndDup, previous_stages=[split_bwa])
+
+    # bam file QC
+    samtools_stat = stage(SamtoolsStats, bam_file=merge_bam_dup.sorted_bam, previous_stages=[merge_bam_dup])
+    samtools_depth = stage(qc.SamtoolsDepth, bam_file=merge_bam_dup.sorted_bam, previous_stages=[merge_bam_dup])
+
+    # variants call via scatter-gather strategy
+    haplotype_caller = stage(SplitHaplotypeCaller, bam_file=merge_bam_dup.sorted_bam, previous_stages=[merge_bam_dup])
+    gather_vcf = stage(GatherVCF, previous_stages=[haplotype_caller])
+
+    # variant filtering
+    select_snps = stage(SelectSNPs, input_vcf=gather_vcf.genotyped_vcf, previous_stages=[gather_vcf])
+    select_indels = stage(SelectIndels, input_vcf=gather_vcf.genotyped_vcf, previous_stages=[gather_vcf])
+    filter_snps = stage(SNPsFiltration, previous_stages=[select_snps])
+    filter_indels = stage(IndelsFiltration, previous_stages=[select_indels])
+
+    # final variant merge
+    merge_vqsr = stage(MergeVariants,
```

Comment:
We're calling this merge_vqsr, but running it on hard-filtered data

Reply:
changed to `merge_vcf`